flytekit/models/task.py

```diff
@@ -100,8 +100,8 @@ def to_flyte_idl(self):
         :rtype: flyteidl.core.tasks_pb2.Resources
         """
         return _core_task.Resources(
-            requests=[r.to_flyte_idl() for r in self.requests],
-            limits=[r.to_flyte_idl() for r in self.limits],
+            requests=[r.to_flyte_idl() for r in self.requests] if self.requests is not None else None,
```

Comment:
we should revert this - locally at least, unit tests passed for me. https://github.com/flyteorg/flytekit/pull/525

Reply:
but now that we can initialize the model with optionally empty requests or limits, we can't, right?

they fail for me

```
FAILED tests/flytekit/unit/models/core/test_workflow.py::test_workflow_template - TypeError: 'NoneType' object is not iterable
FAILED tests/flytekit/unit/models/core/test_workflow.py::test_task_node - TypeError: 'NoneType' object is not iterable
FAILED tests/flytekit/unit/models/core/test_workflow.py::test_node_task_with_no_inputs - TypeError: 'NoneType' object is not iterable
FAILED tests/flytekit/unit/models/core/test_workflow.py::test_node_task_with_inputs - TypeError: 'NoneType' object is not iterable
FAILED tests/flytekit/unit/models/core/test_workflow.py::test_branch_node - TypeError: 'NoneType' object is not iterable
```

Reply:
then why is this passing? https://github.com/flyteorg/flytekit/pull/525